### PR TITLE
fix readme, log, and ctx bugs

### PR DESCRIPTION
**Pull request by lrlna**
- [x] readme example in errors was a bit off
- [x] `server-router` returns `obj.params` and we need to account for that
- [x] add a logName to logger opts
- [x] there is a bug in the logger output

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -135,20 +135,18 @@ Merry comes with a recommended pattern to handle errors.
 // errors.js
 exports.ENOTFOUND = function (req, res, ctx) {
   log.warn('ENOTFOUND')
-  res.statusCode = 404
-  return JSON.stringify({
+  ctx.send(404, { 
     type: 'invalid_request_error',
     message: 'Invalid request data'
-  })
+  })
```
Comment (yoshuawuyts) [inline README.md]: Extra whitespace

File: README.md [hunk 2]
```diff
@@ -135,20 +135,18 @@ Merry comes with a recommended pattern to handle errors.
...
 
 exports.EDBOFFLINE  = function (req, res, ctx) {
   log.error('EDBOFFLINE')
-  res.statusCode = 500 
-  return JSON.stringify({
+  ctx.send(500, { 
     type: 'api_error',
     message: 'Internal server error'
-  })
+  })
```
Comment (yoshuawuyts) [inline README.md]: Extra whitespace

File: README.md [hunk 3]
```diff
@@ -226,6 +224,7 @@ Create a new instance of `merry`. Takes optional opts:
   logging
 - __opts.logStream:__ defaults to `process.stdout`. Set the output writable stream to
   write logs to
+- __opts.logName:__ defaults to `merry`. Sets the name of the logger you're starting
```
Comment (yoshuawuyts) [inline README.md]: I don't think this option is ideal; I find it slightly confusing to name parts of the app, but not the app as a whole. If we're allowing people to name things (and I'm not saying we should), it should probably be a string as the first parameter in the constructor, ("opts" as second), or a "name" key on the "opts" object. In both cases it should still be optional

File: index.js [hunk 4]
```diff
@@ -48,7 +53,7 @@ Merry.prototype.defaultRoute = function (handler) {
 
   function routeHandler (req, res, params) {
     var ctx = new Ctx(req, res, self.log)
-    ctx.params = params
+    ctx.params = params.params
```
Comment (yoshuawuyts) [inline index.js]: We're doing a double object allocation here; I propose we reuse the object returned from the router instead - should save some on GC

File: README.md (hunk 3 above)
Comment (lrlna) [inline README.md]: I'll remove this for now then until we decide on this

File: index.js (hunk 4 above)
Comment (lrlna) [inline index.js]: [this is the object](https://github.com/yoshuawuyts/server-router/blob/master/index.js#L24) that server-router returns, so I have to do `params.params` to actually return the given params. So I am not quite sure what you mean.

**Pull request merged**